src/engine/GeneralCommand.java

```diff
@@ -0,0 +1,86 @@
+package engine;
+
+import data.Data;
+import utils.SortType;
+import utils.Utils;
+
+import java.io.File;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class GeneralCommand<R, D extends Data<R>> {
+    private final D input;
+    private final SortType sortType;
+    private final File outputFile;
+
+    public GeneralCommand(D input, SortType sortType, File outputFile) {
+        this.input = input;
+        this.sortType = sortType;
+        this.outputFile = outputFile;
+    }
+
+    protected String process() {
```

Comment:
Please check access modifiers in this class. 
Do we really need protected here?